gulp/tasks/gen-service-worker.js

```diff
@@ -86,13 +85,29 @@ gulp.task('gen-service-worker', () => {
         rootDir + '/fonts/roboto/Roboto-Bold.ttf',
         rootDir + '/images/card_media_player_bg.png',
       ],
-      runtimeCaching: [{
-        urlPattern: /\/static\/translations\//,
+      // Rules are proceeded in order and negative per-domain rules are not supported.
+      runtimeCaching: [
+        { // Cache translations on first access
+          urlPattern: '/static/translations/*',
         handler: 'cacheFirst',
-      }, {
-        urlPattern: RegExp('^[^/]*/' + fallbackList + '.'),
+        },
+        { // Get static, api, and local (and home-assistant-polymer in dev mode) from network.
```

Comment:
why would we go for network? We should get these from cache. Fingerprints allow us to make sure that these are always the same.

Reply:
You mean `/static`?

We don't fingerprint everything:
There are fonts, icons, pagedown-js

That said, when SW updates cache is cleared. This will happen if any pre-cache hash is changed.

Reply:
Fonts/Icons/pagedown-js are fine if they get old versions.

In my opinion, the goal of the service worker should be that only data transferred between phone and server is the websocket connection and potentially some image requests.

This is how it currently looks:

![screen shot 2017-11-05 at 10 15 44](https://user-images.githubusercontent.com/1444314/32417704-6eeef6f2-c212-11e7-8e30-031ad51f2317.png)

Reply:
Made me realize btw that we should move themes into the websocket API

Reply:
Also the fonts should be loaded before the panels.

Reply:
(not for this PR btw)

Reply:
I'm fine with not caching local at all in the service worker.

Reply:
Updated rules